druid/src/widget/stepper.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright 2019 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! A stepper widget.
+
+use crate::{
+    BoxConstraints, Env, Event, EventCtx, LayoutCtx, PaintCtx, Size, TimerToken, UpdateCtx, Widget,
+};
+use std::f64::EPSILON;
+use std::time::{Duration, Instant};
+
+use crate::kurbo::{BezPath, Rect, RoundedRect};
+use crate::piet::{LinearGradient, RenderContext, UnitPoint};
+
+use crate::theme;
+use crate::widget::Align;
+use crate::Point;
+
+/// A stepper widget for step-wise increasing and decreasing a value.
+pub struct Stepper {
+    max: f64,
+    min: f64,
+    step: f64,
+    wrap: bool,
+    /// A closure that will be invoked when the value changed.
+    value_changed: Box<dyn Fn(&mut EventCtx, &mut f64, &Env)>,
+    /// Keeps track of which button is currently triggered.
+    increase_active: bool,
+    decrease_active: bool,
+    timer_id: TimerToken,
+}
+
+impl Stepper {
+    pub fn new(
```

Comment:
I might consider splitting some of these fields into builder methods. By default I might only require the `value_changed` argument; other arguments could be optional? I think that reasonable defaults exists (`std::f64::MIN`, `std::f64::MAX`, `1.0`, `false`)?